### PR TITLE
Forgot black and some features so did it again

### PR DESCRIPTION
**Pull request by preet4737**

### CODE REVIEW

File: canteenWeb/models.py [hunk 1]
```diff
@@ -7,15 +7,35 @@
...
+    )  # to check whether the user is teacher
+    is_banned = models.BooleanField(
+        default=False
+    )  # To check whether the user is banned
+    is_staff = models.BooleanField(default=False)
 
 
 class StudentProfile(models.Model):
-    pass
+    student = models.OneToOneField(User, on_delete=models.CASCADE, null=True)
```
Comment (vixrant) [inline canteenWeb/models.py]: Change this to user instead of student.

File: canteenWeb/models.py [hunk 2]
```diff
@@ -7,15 +7,35 @@
...
+    division = models.CharField(max_length=1, default="A")
+    department = models.CharField(max_length=30, default=None)
+    admission_year = models.PositiveSmallIntegerField(default=1)
 
 
 class TeacherProfile(models.Model):
-    pass
+    teacher = models.OneToOneField(User, on_delete=models.CASCADE, null=True)
```
Comment (vixrant) [inline canteenWeb/models.py]: Change this to user instead of teacher.

File: canteenWeb/models.py [hunk 3]
```diff
@@ -7,15 +7,35 @@
...
+    floor = models.PositiveSmallIntegerField(default=None)
+    department = models.CharField(max_length=30, default=None)
```
Comment (vixrant) [inline canteenWeb/models.py]: Don't set default=None everywhere. This is a required field.

File: canteenWeb/models.py [hunk 4]
```diff
@@ -7,15 +7,35 @@
...
+    )  # To check whether the user is banned
+    is_staff = models.BooleanField(default=False)
 
 
 class StudentProfile(models.Model):
-    pass
+    student = models.OneToOneField(User, on_delete=models.CASCADE, null=True)
+    division = models.CharField(max_length=1, default="A")
+    department = models.CharField(max_length=30, default=None)
+    admission_year = models.PositiveSmallIntegerField(default=1)
```
Comment (vixrant) [inline canteenWeb/models.py]: Remove default.

File: canteenWeb/models.py [hunk 5]
```diff
@@ -7,15 +7,35 @@
 
 
 class User(AbstractUser):
-    pass
+    USERNAME_FIELD = models.CharField(max_length=30, unique=True, default="")  # sap id
+    EMAIL_FIELD = models.EmailField(max_length=40, blank=True, unique=True)
+    first_name = models.CharField(max_length=15)
+    last_name = models.CharField(max_length=15)
```
Comment (vixrant) [inline canteenWeb/models.py]: Remove USERNAME_FIELD, EMAIL_FIELD, first_name, and last_name

File: canteenWeb/models.py (hunk 4 above)
Comment (preet4737) [reply canteenWeb/models.py]: It says these are non nullable fields so I had to add default

File: canteenWeb/models.py (hunk 5 above)
Comment (preet4737) [reply canteenWeb/models.py]: Why ? USERNAME_FIELD is for sap id right as we discussed.Give me the reason why you want to remove it .

File: canteenWeb/models.py (hunk 3 above)
Comment (preet4737) [reply canteenWeb/models.py]: While migrating they ask for default as it is non nullable field

File: canteenWeb/models.py (hunk 5 above)
Comment (vixrant) [reply canteenWeb/models.py]: That I wrote with respect to AbstractBaseUser model. No need in AbstractUser model. Here are the differences: https://simpleisbetterthancomplex.com/tutorial/2016/07/22/how-to-extend-django-user-model.html

File: canteenWeb/models.py (hunk 1 above)
Comment (preet4737) [reply canteenWeb/models.py]: Okay...

File: canteenWeb/models.py (hunk 4 above)
Comment (vixrant) [reply canteenWeb/models.py]: See `student` field (rename to `user` field) should not be null.
`division` default is okay.
`department` default is blank. Set blank=True.
`admission_year` default = 1 doesn't make sense. What is year = 1? Set it as null=True.

File: canteenWeb/forms.py [hunk 6]
```diff
@@ -1,5 +1,6 @@
 from django import forms
 
+
 class UserForm(forms.ModelForm):
```
Comment (sahiljajodia01) [inline canteenWeb/forms.py]: What are these forms for?

Comment (vixrant) [reply canteenWeb/forms.py]: For Django Admin. That requires Forms.

Comment (vixrant) [reply canteenWeb/forms.py]: But he's still working on integrating our custom user model with Django Admin.

**Pull request merged**